libs/src/service-selection/ServiceSelection.test.js

```diff
@@ -121,6 +122,35 @@ describe('ServiceSelection', () => {
     const service = await s.select()
     assert.strictEqual(service, slow)
   })
+
+  it('will recheck unhealthy ones', async () => {
+    const atFirstHealthy = 'https://atFirstHealthy.audius.co'
+    nock(atFirstHealthy)
+      .get('/health_check')
+      .reply(200)
+    nock(atFirstHealthy)
+      .get('/health_check')
+      .reply(400)
+
+    const atFirstUnhealthy = 'https://atFirstUnhealthy.audius.co'
+    nock(atFirstUnhealthy)
+      .get('/health_check')
+      .reply(400)
+    nock(atFirstUnhealthy)
+      .get('/health_check')
+      .reply(200)
+
+    const s = new ServiceSelection({
+      getServices: () => [atFirstHealthy, atFirstUnhealthy],
+      unhealthyTTL: 0
+    })
+    const firstService = await s.select()
+    assert.strictEqual(firstService, atFirstHealthy)
+
+    await Utils.wait(200)
```

Comment:
maybe i missed something but how does waiting 200ms change the result of the select() call? looks like the /health_check endpoints get overridden with their second  nock status code calls

Reply:
just need this to allow the unhealthy list to be cleared.

the first calls go out and atFirstHealthy is the provider that's picked. at setTimeout(0),
because that's at the end of the event look there needs to be some wait for that to get cleared. adding a comment.